### PR TITLE
With syslog

### PR DESCRIPTION
**Pull request by cviecco**

### CODE REVIEW

**Comment by rgooch** [comment]
I don't like this approach. I would prefer that you use a teelogger and pass that in when creating the AWS Broker. That will avoid the need for code changes and API changes.

**Comment by rgooch** [comment]
Following on from our discussion (passing in two loggers) please call this something like `auditLogger` with type `lib/log/Logger` so that the API is more abstract. The `auditLogger` should be a teelogger so that you only need to make one logging call (it will go to both the syslog and regular logger).
You can create the teelogger like this:
`teeLogger := teelogger.New(logger, debuglogger.Upgrade(log.New(syslog.New(syslog.LOG_AUTHPRIV|syslog.LOG_NOTICE, "cloud-gate"), "", 0))`

File: broker/aws/api.go [hunk 1]
```diff
@@ -34,10 +35,11 @@ type Broker struct {
 
 func New(userInfo userinfo.UserInfo,
 	credentialsFilename string,
-	logger log.DebugLogger) *Broker {
+	logger log.DebugLogger, auditLog log.DebugLogger) *Broker {
```
Comment (rgooch) [inline broker/aws/api.go]: s/auditLog/auditLogger/

File: broker/aws/impl.go [hunk 2]
```diff
@@ -438,7 +438,8 @@ func (b *Broker) getConsoleURLForAccountRole(accountName string, roleName string
 	targetUrl := fmt.Sprintf("%s?Action=login&Issuer=https://example.com&Destination=%s&SigninToken=%s", federationUrl, awsDestinationURL, tokenOutput.SigninToken)
 	b.logger.Debugf(1, "targetURL=%s", targetUrl)
 
-	b.logger.Printf("Cosole url generated for: %s on account %s role %s", userName, accountName, roleName)
+	logString := fmt.Sprintf("Console url generated for: %s on account %s role %s", userName, accountName, roleName)
+	b.auditLogger.Printf("%s", logString)
```
Comment (rgooch) [inline broker/aws/impl.go]: Please collapse into a single line.

File: broker/aws/impl.go [hunk 3]
```diff
@@ -463,7 +464,8 @@ func (b *Broker) generateTokenCredentials(accountName string, roleName string, u
 		SessionToken: *assumeRoleOutput.Credentials.SessionToken,
 		Region:       region,
 	}
-	b.logger.Printf("Token credentials generated for: %s on account %s role %s", userName, accountName, roleName)
+	logString := fmt.Sprintf("Token credentials generated for: %s on account %s role %s", userName, accountName, roleName)
+	b.auditLogger.Printf("%s", logString)
```
Comment (rgooch) [inline broker/aws/impl.go]: Please collapse into a single line.

File: cmd/cloud-gate/main.go [hunk 4]
```diff
@@ -29,6 +33,13 @@ func main() {
 	}
 	logger := serverlogger.New("")
 
+	syslogWriter, err := syslog.New(syslog.LOG_AUTHPRIV|syslog.LOG_NOTICE, "cloud-gate")
+	if err != nil {
+		logger.Printf("Cound not open connection to local syslog daemon")
```
Comment (rgooch) [inline cmd/cloud-gate/main.go]: s/Cound/Could/

Comment (cviecco) [reply cmd/cloud-gate/main.go]: done

Comment (cviecco) [reply cmd/cloud-gate/main.go]: done

File: broker/aws/impl.go (hunk 2 above)
Comment (cviecco) [reply broker/aws/impl.go]: done

**Pull request merged**